finder-app/writer.c

```diff
@@ -0,0 +1,47 @@
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <syslog.h>
+#include <unistd.h>
+#include <string.h>
+
+int main(int argc , char *argv[]){
+	
+
+	openlog(NULL, 0, LOG_USER);
+	
+	if(argc != 3){
+		printf("Invalid number of arguments, Please enter required 2 arguments (<filename> <write string>).\n\r");
+		syslog(LOG_ERR,"Invalid number of arguments.");
+		return 1;
+	}
+	
+	
+	int fd=open(argv[1], O_WRONLY | O_CREAT ,S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH);
+
+	if(fd == -1){
+		syslog(LOG_ERR,"The file %s is not avaliable or created.", argv[1]);
+		return 1;
+	}
+	
+	int error = write(fd,argv[2],strlen(argv[2]));
```

Comment:
@akshithaluguri22 I suggest using a variable for these to decouple from argument design and improve readability.